### PR TITLE
Fix partial response messages for nested frame setup

### PR DESCRIPTION
**Pull request by Kanac**
Related PR https://github.com/OfficeDev/microsoft-teams-library-js/pull/459

The above PR introduces the notion of partial responses. This is for Teams SDK calls that require more than 1 response from Teams to fully complete the request (i.e `getMedia`). However, there is a problem when there are nested frames that are both using the Teams SDK. If the inner frame uses an API like `getMedia` which uses partial responses, it doesn't quite work. 

When the outer frame passes the partial response from Teams back to the inner frame, the response is mistakenly not formatted as a partial response for the inner frame. This causes APIs like `getMedia` to terminate early before the whole response is given back to the inner frame. This PR fixes this issue by passing `isPartialResponse` to the end of the argument list to the response proxy callbacks so that it could be used for formatting the proxy response as partial.

Method Response Chain **Before** (for calls like `getMedia`): 
Teams -> (Sends Partial Response) -> Outer Frame using Teams SDK -> (Sends **Non-Partial** Response) -> Inner Frame using Teams SDK

Method Response Chain **After** (for calls like `getMedia`): 
Teams -> (Sends Partial Response) -> Outer Frame using Teams SDK -> (Sends **Partial** Response) -> Inner Frame using Teams SDK

### CODE REVIEW

File: test/utils.ts [hunk 1]
```diff
@@ -49,20 +51,20 @@ export class Utils {
       outerHeight: 768,
       screenLeft: 0,
       screenTop: 0,
-      addEventListener: function (type: string, listener: (ev: MessageEvent) => void, useCapture?: boolean): void {
```
Comment (Kanac) [inline test/utils.ts:52]: Most of the changes here are formatting fixes from my IDE, but they seem to make the code cleaner. I could remove these if there's any issues though.

**Comment by Kanac** [comment]
@ydogandjiev

**Review by ydogandjiev — APPROVED** [review]

File: src/internal/internalAPIs.ts [hunk 2]
```diff
@@ -235,7 +235,7 @@ export function handleParentMessage(evt: DOMMessageEvent): void {
     const message = evt.data as MessageResponse;
     const callback = GlobalVars.callbacks[message.id];
     if (callback) {
-      callback.apply(null, message.args);
+      callback.apply(null, [...message.args, message.isPartialResponse]);
```
Comment (ydogandjiev) [inline src/internal/internalAPIs.ts:238]: Why is this change necessary? Seems weird that we would be passing this property to the callback? I don't think this is something a callback should ever have to handle itself.

**Review by ydogandjiev — CHANGES_REQUESTED** [review]

File: src/internal/internalAPIs.ts (hunk 2 above)
Comment (Kanac) [reply src/internal/internalAPIs.ts:238]: Its used specifically for this part here where we proxy the response from Teams to the current frame to the child frame that called the API https://github.com/OfficeDev/microsoft-teams-library-js/pull/483/files#diff-fb785af0e32520a03c689ec3c20c481d2c9b5d362ef181402123f2b50be7a418R277

I don't really see any other better way to pass `isPartialResponse` to reconstructing the proxy partial response to the child frame so that's why this change is necessary.

**Comment by SonalJha** [comment]
Changes look good to me. Please test with mobile (android and ios) before merging the PR.

**Review by SonalJha — APPROVED** [review]

File: src/internal/internalAPIs.ts (hunk 2 above)
Comment (Kanac) [reply src/internal/internalAPIs.ts:238]: One thing we can do to be safer is to only pass `isPartialResponse` to the callback args if the callback is specificially the proxy callback defined here https://github.com/OfficeDev/microsoft-teams-library-js/pull/483/files#diff-fb785af0e32520a03c689ec3c20c481d2c9b5d362ef181402123f2b50be7a418L274 
(i.e we can change it from an anonymous function to a named one, and check the callbacks name).
But either way, given how the current code works, passing this property as an arg to the callback seems to be the only way to solve this issue.

**Review by ydogandjiev — APPROVED** [review]

File: src/internal/internalAPIs.ts (hunk 2 above)
Comment (ydogandjiev) [reply src/internal/internalAPIs.ts:238]: I'm fine with either option; I would say do what you think is best. FYI @sachingupta

File: src/internal/interfaces.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ export interface MessageRequest {
 export interface MessageResponse {
   id: number;
   args?: any[]; // tslint:disable-line:no-any The args here are a passthrough from OnMessage where we do receive any[]
+  isPartialResponse?: boolean;
```
Comment (sachingupta) [inline src/internal/interfaces.ts:40]: add some detail about this field.

**Review by sachingupta — APPROVED** [review]
